include/engine/routing_algorithms/shortest_path.hpp

```diff
@@ -114,7 +114,11 @@ class ShortestPathRouting final
                           needs_loop_forwad,
                           needs_loop_backwards);
         }
-        new_total_distance += std::min(total_distance_to_forward, total_distance_to_reverse);
+        //if no route is found between two parts of the via-route, the entire route becomes invalid
+        if( new_total_distance == INVALID_EDGE_WEIGHT )
```

Comment:
That was my first intuition as well. It turns out, the way it is used here is different. If you have a look at the function header, the two weights are constants passed down here. New total distance on the other hand is used to pass down to the search functions as distance value.

This could probably use a serious refactor for better names, but I didn't want to mix this up with the actual fix here.